vllm/model_executor/models/h2ovl.py

```diff
@@ -0,0 +1,400 @@
+            (i // (target_width // image_size)) * image_size,
+            ((i % (target_width // image_size)) + 1) * image_size,
+            ((i // (target_width // image_size)) + 1) * image_size,
+        )
+        # split the image
+        split_img = resized_img.crop(box)
+        processed_images.append(split_img)
+    assert len(processed_images) == blocks
+    if use_thumbnail and len(processed_images) != 1:
+        thumbnail_img = image.resize((image_size, image_size))
+        processed_images.append(thumbnail_img)
+    return processed_images
+
+
+def load_image1(image: Image.Image, input_size=448, min_num=1, max_num=6):
+    # image = Image.open(image_file).convert('RGB')
+    transform = build_transform(input_size=input_size)
+    images, target_aspect_ratio = dynamic_preprocess(
+        image,
+        image_size=input_size,
+        use_thumbnail=True,
+        min_num=min_num,
+        max_num=max_num,
+    )
+    pixel_values = [transform(image) for image in images]
+    pixel_values = torch.stack(pixel_values)
+    return pixel_values, target_aspect_ratio
+
+
+def load_image2(
+    image: Image.Image,
+    input_size=448,
+    min_num=1,
+    max_num=6,
+    target_aspect_ratio=None,
+):
+    # image = Image.open(image_file).convert('RGB')
+    transform = build_transform(input_size=input_size)
+    images = dynamic_preprocess2(
+        image,
+        image_size=input_size,
+        use_thumbnail=True,
+        min_num=min_num,
+        max_num=max_num,
+        prior_aspect_ratio=target_aspect_ratio,
+    )
+    pixel_values = [transform(image) for image in images]
+    pixel_values = torch.stack(pixel_values)
+    return pixel_values
+
+
+def image_to_pixel_values(
+    image: Image.Image,
+    input_size: int,
+    min_num: int,
+    max_num: int,
+    use_thumbnail: bool,
+    use_MSAC: bool,
+) -> torch.Tensor:
+    # when MSAC is turned on, we need to process the image twice
+    if use_MSAC:
+        pixel_values, target_aspect_ratio = load_image1(image,
+                                                        input_size=input_size,
+                                                        min_num=min_num,
+                                                        max_num=max_num)
+        pixel_values2 = load_image2(
+            image,
+            input_size=input_size,
+            min_num=min_num,
+            max_num=max_num,
+            target_aspect_ratio=target_aspect_ratio,
+        )
+        pixel_values = torch.cat(
+            [pixel_values2[:-1], pixel_values[:-1], pixel_values2[-1:]], 0)
+
+    else:
+        transform = build_transform(input_size=input_size)
+        images, _ = dynamic_preprocess(
+            image,
+            min_num=min_num,
+            max_num=max_num,
+            image_size=input_size,
+            use_thumbnail=use_thumbnail,
+        )
+        pixel_values = [transform(image) for image in images]
+        pixel_values = torch.stack(pixel_values)
```

Comment:
This seems like the duplicate implementation of `load_image1`. Why can't we just call `load_image1(...)` here?